src/resources/defi/utils.ts

```diff
@@ -3,87 +3,181 @@ import {
 import { maybeSignUri } from '@/handlers/imgix';
 import { ethereumUtils } from '@/utils';
 import { chainsIdByName } from '@/chains';
 
-export const parsePosition = (position: Position, currency: NativeCurrencyKey): RainbowPosition => {
-  let totalLocked = '0';
+const PROTOCOL_VERSION_REGEX = /-[vV]\d+$/;
+const LP_POOL_SYMBOL = 'LP-POOL';
+const CONCENTRATED_LIQUIDITY_ONLY_DAPPS = ['uniswap-v3'];
 
-  let totalDeposits = '0';
-  const parsedDeposits = position.deposits?.map((deposit): RainbowDeposit => {
+// Zerion currently has no specific lp position signifier
+function isLpPositionItem(item: Deposit | Stake): boolean {
+  return item.asset.symbol === LP_POOL_SYMBOL;
+}
+
+function calculatePositionItemNativeDisplayValue(item: UnderlyingAsset | Claimable | Stake, currency: NativeCurrencyKey) {
+  const decimals = typeof item.asset.decimals === 'number' ? item.asset.decimals : 18;
+  return convertRawAmountToNativeDisplay(item.quantity, decimals, item.asset.price?.value!, currency);
+}
+
+function addPositionTotals(totals1: PositionsTotals, totals2: PositionsTotals, currency: NativeCurrencyKey): PositionsTotals {
+  return {
+    totals: {
+      amount: add(totals1.totals.amount, totals2.totals.amount),
+      display: convertAmountToNativeDisplay(add(totals1.totals.amount, totals2.totals.amount), currency),
+    },
+    totalLocked: add(totals1.totalLocked, totals2.totalLocked),
+    borrows: {
+      amount: add(totals1.borrows.amount, totals2.borrows.amount),
+      display: convertAmountToNativeDisplay(add(totals1.borrows.amount, totals2.borrows.amount), currency),
+    },
+    claimables: {
+      amount: add(totals1.claimables.amount, totals2.claimables.amount),
+      display: convertAmountToNativeDisplay(add(totals1.claimables.amount, totals2.claimables.amount), currency),
+    },
+    deposits: {
+      amount: add(totals1.deposits.amount, totals2.deposits.amount),
+      display: convertAmountToNativeDisplay(add(totals1.deposits.amount, totals2.deposits.amount), currency),
+    },
+    stakes: {
+      amount: add(totals1.stakes.amount, totals2.stakes.amount),
+      display: convertAmountToNativeDisplay(add(totals1.stakes.amount, totals2.stakes.amount), currency),
+    },
+  };
+}
+
+/**
+ * Parses a pool string to extract token symbols
+ * @param {string} name - String in format "Name TOKEN1/TOKEN2 Pool (#ID)"
+ * @returns {string[]} Array of token symbols
+ */
+function parseTokenSymbolsFromPoolName(name: string): string[] {
+  try {
+    const match = name.match(/\b[A-Z]+(?:\/[A-Z]+)+\b/);
+    if (!match) {
+      return [];
+    }
+    return match[0].split('/');
+  } catch (error) {
+    return [];
+  }
+}
+
+// hack to fix for Zerion api returning only 1 asset for lp position when position is out of range
+function fixOutOfRangeLpPosition(positionItem: Deposit | Stake, claimables: Claimable[]): Deposit | Stake {
```

Comment:
do we have a conversation opened with them about fixing this? Just curious

Reply:
Yes we do, looks like it might be a while before it's fixed